src/wallet/walletdb.cpp

```diff
@@ -749,11 +749,6 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
             }
             wss.fIsEncrypted = true;
         }
-        else if (strType == "hdchain")
-        {
-            auto chain = CHDChain::Read(ssValue);
-            pwallet->SetLegacyHDChain(chain, true);
-        }
```

Comment:
I don't understand what this is well enough to know whether it's safe to drop it. In general I'm unhappy with the sparse documentation of the wallet format. ZIP 401 [says](https://zips.z.cash/zip-0400#schema) that it is the chain code derived from the hd seed, but that is trivial to compute so why was it ever stored?

Reply:
The CHDChain value also stored the Sapling account counter to indicate the next available account id, along with the fingerprint. That had to be updated every time `z_getnewaddress` was called (since that badly-named method actually creates a whole separate spend authority.)